server/api.js

```diff
@@ -63,6 +63,107 @@ router.get("/", (_, res) => {
 	res.status(200).json({ message: "WELCOME TO LOVE ME TENDER SITE" });
 });
 
+function generateRandomPassword(length = 8) {
+	const lowerCase = "abcdefghijklmnopqrstuvwxyz";
+	const upperCase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+	const numbers = "0123456789";
+	const specialChars = "!@#$%^&*()_-+=";
+
+	const allChars = lowerCase + upperCase + numbers + specialChars;
+
+	let password = "";
+	password += lowerCase.charAt(Math.floor(Math.random() * lowerCase.length));
+	password += upperCase.charAt(Math.floor(Math.random() * upperCase.length));
+	password += numbers.charAt(Math.floor(Math.random() * numbers.length));
+	password += specialChars.charAt(
+		Math.floor(Math.random() * specialChars.length)
+	);
+
+	for (let i = 4; i < length; i++) {
+		password += allChars.charAt(Math.floor(Math.random() * allChars.length));
+	}
+
+	password = password
+		.split("")
+		.sort(() => 0.5 - Math.random())
+		.join("");
+
+	return password;
+}
+
+router.post("/signup", async (req, res) => {
+	const {
+		email,
+		userType,
+		firstName,
+		lastName,
+		userName,
+		company,
+		address,
+		description,
+	} = req.body;
+
+	if (!email || !userType || !firstName || !lastName) {
```

Comment:
Do all validations before doing any queries and return all errors found in an array